NitroxServer/Communication/Packets/Processors/PlayerDeathEventProcessor.cs

```diff
@@ -17,9 +17,10 @@ public PlayerDeathEventProcessor(PlayerManager playerManager, ServerConfig serve
 
         public override void Process(PlayerDeathEvent packet, Player player)
         {
-            if(serverConfig.GameMode == "Hardcore")
+            if(serverConfig.IsGameMode("Hardcore"))
             {
                 player.IsPermaDeath = true;
+                playerManager.UpdatePlayer(player);
```

Comment:
Is this required?  I think the player object already comes from PlayerManager: https://github.com/SubnauticaNitrox/Nitrox/blob/master/NitroxServer/Communication/Packets/PacketHandler.cs#L27

Reply:
Possibly not but I found if permadeath was updated on the player to ispermadeath = true then it was never being persistantly set as the playermanager list was never updated

Reply:
I think there might be another bug at play here then.  We have other packet processors that update the player object.  So if the persisted information is using a different reference then we should track down what is going on.  Can you test once again to verify that this is the case?  I spot checked the code and the reference looks to be the same.

Reply:
Recheck you are correct this was due to earlier when the player death wasnt being handled properly